onnxruntime/test/python/quantization/test_save_activations.py

```diff
@@ -0,0 +1,164 @@
+        relu_node_1 = helper.make_node("Relu", ["input"], ["X1"], name="Relu1")
+        conv_node_1 = helper.make_node("Conv", ["X1", "W1", "B1"], ["X2"], name="Conv1")
+        relu_node_2 = helper.make_node("Relu", ["X2"], ["X3"], name="Relu2")
+        conv_node_2 = helper.make_node("Conv", ["X3", "W3", "B3"], ["X4"], name="Conv2")
+        conv_node_3 = helper.make_node("Conv", ["X1", "W5", "B5"], ["X5"], name="Conv3")
+        add_node = helper.make_node("Add", ["X4", "X5"], ["X6"], name="Add")
+
+        # we are keeping all tensors in the output anyway for verification purpose
+        graph = helper.make_graph(
+            [relu_node_1, conv_node_1, relu_node_2, conv_node_2, conv_node_3, add_node],
+            "test_graph_4",
+            [input],
+            [x1_output, x2_output, x3_output, x4_output, x5_output, x6_output],
+        )
+        graph.initializer.add().CopyFrom(w1)
+        graph.initializer.add().CopyFrom(b1)
+        graph.initializer.add().CopyFrom(w3)
+        graph.initializer.add().CopyFrom(b3)
+        graph.initializer.add().CopyFrom(w5)
+        graph.initializer.add().CopyFrom(b5)
+        model = helper.make_model(graph, opset_imports=[helper.make_opsetid("", 13)])
+        onnx.save(model, test_model_path)
+
+    def test_saved_act(self):
+        test_model_path = str(Path(self._tmp_model_dir.name).joinpath("test_model_1.onnx"))
+        self.construct_test_model1(test_model_path)
+        data_reader = TestDataReader()
+
+        aug_model = aug_model_for_act_saving(test_model_path)
+        augmented_model_path = str(Path(self._tmp_model_dir.name).joinpath("augmented_test_model_1.onnx"))
+
+        onnx.save(
+            aug_model,
+            augmented_model_path,
+            save_as_external_data=False,
+        )
+
+        tensor_dict = run_collect_activations(augmented_model_path, data_reader)
+
+        # run original model and compare the tensors
+        sess_options = onnxruntime.SessionOptions()
+        sess_options.graph_optimization_level = onnxruntime.GraphOptimizationLevel.ORT_DISABLE_ALL
+        infer_session = onnxruntime.InferenceSession(
+            test_model_path,
+            sess_options=sess_options,
+            providers=["CPUExecutionProvider"],
+        )
+
+        data_reader.rewind()
+        oracle_outputs = []
+        while True:
+            input = data_reader.get_next()
+            if not input:
+                break
+            oracle_outputs.append(infer_session.run(None, input))
+
+        output_dict={}
+        output_info = infer_session.get_outputs();
+        for batch in oracle_outputs:
+            for output, output_data in zip(output_info, batch):
+                output_dict.setdefault(output.name, []).append(output_data)
+        
+        for outputn, outputlist in output_dict.items():
+            test_output_list = tensor_dict.get(outputn)
+            for expected, actual in zip(outputlist, test_output_list):
```

Comment:
naming Prefer using plurals, e.g. `test_outputs` to name container variables

Reply:
fixed